### PR TITLE
[move] High-limit txn fixes

### PR DESCRIPTION
**Pull request by georgemitenkov**
## Description

Multiple fixes around the VM / Move code for high-limit txn features and versioned Move prologue/epilogues.

1. Versioned prologue now correctly handles the case where a sender can be forced to pay gas for an open sponsored transaction. This is ensured by making sure we use `Option<signer>` for fee payer. If `None`, there is no fee payer flow used, if `Some` - fee payer authentification runs. Additionally, added sender and fee payer to `PrologueArgs`/`EpilogueArgs` so that signatures do not need to change in the future at all.
2. Changed `transaction_limits::new_tier` to be `friend`.
3. `find_min_stake_required` - kept as linear search, number of tiers is expected to be small so linear scan in Move will be faster. Can be revisited later if things change.
4. Approved governance script cannot carry a staking limits request - VM will abort with `TXN_LIMITS_REQUEST_NOT_ALLOWED_FOR_GOVERNANCE_SCRIPT` status code.
5. Adding gating for `SLH_DSA_SHA2_128S_SIGNATURE` and other signature checks to execution path in addition to mempool.
6. Fixing sponsored automatic account creation that skips sender auth-key validation when `DEFAULT_ACCOUNT_RESOURCE` is disabled.

## How Has This Been Tested?

Fee payer tests + txn limits tests.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

---

> [!NOTE]
> **High Risk**
> Touches transaction authentication/prologue validation and high-limit staking-based limit enforcement, where subtle logic or gating mistakes can lead to invalid transactions being accepted/rejected incorrectly. Also updates on-chain/BSC-serialized structs (multiplier field rename) and status code mappings, which can cause compatibility issues if not coordinated.
> 
> **Overview**
> Fixes high-limit transaction limits by switching requested multipliers from basis-points naming to *percent-of-base* (`execution_multiplier_percent`/`io_multiplier_percent`) across Rust types, gas metering, Move `transaction_limits`, and the generated-format YAMLs.
> 
> Tightens high-limit request validation: stake-pool/delegation-pool requests now require the pool to be in the current validator set (new `STAKE_POOL_NOT_IN_VALIDATOR_SET` mapping), and approved governance scripts are rejected if they attach a staking limits request (new `TXN_LIMITS_REQUEST_NOT_ALLOWED_FOR_GOVERNANCE_SCRIPT`).
> 
> Hardens transaction validation paths by centralizing authenticator feature gating (`SingleSender`, `WebAuthn`, `SLH_DSA`) and invoking it in both mempool validation and execution, plus fixes sponsored/fee-payer prologue argument handling by adding an explicit `needs_fee_payer_auth_check` flag and correcting sender auth-key validation behavior when sponsored auto-account-creation applies. Adds/updates e2e tests covering these edge cases and improves prologue error reporting to include the unexpected error message.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0ff0538705a1f5df74adf1e352b4e2401ae7924d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by georgemitenkov** [comment]
* **#19725** <a href="https://app.graphite.com/github/pr/aptos-labs/aptos-core/19725?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/aptos-labs/aptos-core/19725?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -3497,11 +3533,8 @@ impl VMValidator for AptosVM {
             return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING);
         }
 
-        let txn = match transaction.check_signature() {
-            Ok(t) => t,
-            _ => {
-                return VMValidatorResult::error(StatusCode::INVALID_SIGNATURE);
-            },
```
[the remark this answers is not in the document]
Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs:3510]: Indeed, accidnetal, fixed

File: aptos-move/framework/aptos-framework/sources/transaction_limits.move [hunk 2]
```diff
@@ -68,13 +70,14 @@ module aptos_framework::transaction_limits {
...
 
-    /// Multipliers requested by the user, expressed in basis points (That is,
-    /// 1x is 100, 2.5x is 250).
+    /// Multipliers requested by the user.
     ///
     /// INVARIANT: must match Rust enum for BCS serialization.
     enum RequestedMultipliers has copy, drop, store {
         V1 {
+            /// Execution-gas multiplier (100 is 1x).
             execution_bps: u64,
```
Comment (igor-aptos) [inline aptos-move/framework/aptos-framework/sources/transaction_limits.move]: what is bps?

in general basis points are 1 / 10000, so 100bps is 1%

if this is percentage, we should name it as such?

Comment (georgemitenkov) [reply aptos-move/framework/aptos-framework/sources/transaction_limits.move]: Basis points, but scaled to 100 mean 1x (named it to reflect 1/100, not 1/100 of a percent). Percentage seems wrong as this is really a scaling factor. Open to name suggestions :)

Comment (igor-aptos) [reply aptos-move/framework/aptos-framework/sources/transaction_limits.move]: I don't think we should redefine what bps is:

> A basis point (abbreviated as bps, often pronounced "bips") is a unit of measure equal to one-hundredth of one percentage point

maybe `execution_multiplier_percent`, unless you think of something shorter

File: aptos-move/framework/aptos-framework/sources/transaction_validation.move [hunk 3]
```diff
@@ -148,16 +149,21 @@ module aptos_framework::transaction_validation {
...
+                        !account::exists_at(sender_address) &&
+                        features::sponsored_automatic_account_creation_enabled()
                 ) {
-                    assert!(
-                        txn_authentication_key == option::some(account::get_authentication_key(sender_address)),
-                        error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
-                    );
+                    // This is a sponsored transaction with account that does
+                    // not exist and there is no default account resource.
+                    bcs::to_bytes(&sender_address)
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/transaction_validation.move:159]: this looks unrelated change?

**Review by zekun000 — COMMENTED** [review]
are we addressing this one as well? 
```
Staking-based limits do not require the stake pool / delegation pool to be active or meaningfully locked.
```

File: aptos-move/framework/aptos-framework/sources/transaction_validation.move (hunk 3 above)
Comment (georgemitenkov) [reply aptos-move/framework/aptos-framework/sources/transaction_validation.move:159]: See `test_sponsored_new_account_sender_auth_key_not_skipped_when_default_account_resource_disabled` - this is an existing issue if default resource is disabled (which is not, but surfaced during review and we better fix it in case some feature configs change)

File: aptos-move/framework/aptos-framework/sources/transaction_limits.move (hunk 2 above)
Comment (georgemitenkov) [reply aptos-move/framework/aptos-framework/sources/transaction_limits.move]: Renamed to `execution_multiplier_percent` and similar `_percent`, etc. throughout

**Review by zekun000 — APPROVED** [review]

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 4]
```diff
@@ -2305,6 +2345,7 @@ impl AptosVM {
             &'a C,
         ) -> G,
     {
+        self.check_authenticator_features(txn.authenticator_ref())?;
```
[the remark this answers is not in the document]
Comment (georgemitenkov) [reply aptos-move/aptos-vm/src/aptos_vm.rs:2348]: Irrelevant for this PR
